SingularityService/src/main/java/com/hubspot/singularity/SingularityAuthModule.java

```diff
@@ -1,27 +1,33 @@
 package com.hubspot.singularity;
 
-import com.google.common.base.Optional;
 import com.google.inject.Binder;
-import com.google.inject.Module;
 import com.google.inject.Scopes;
-import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.Multibinder;
+import com.hubspot.dropwizard.guicier.DropwizardAwareModule;
+import com.hubspot.singularity.auth.SingularityAuthFeature;
+import com.hubspot.singularity.auth.SingularityAuthenticatorClass;
 import com.hubspot.singularity.auth.SingularityAuthorizationHelper;
 import com.hubspot.singularity.auth.authenticator.SingularityAuthenticator;
+import com.hubspot.singularity.auth.authenticator.SingularityMultiMethodAuthenticator;
 import com.hubspot.singularity.auth.datastore.SingularityAuthDatastore;
 import com.hubspot.singularity.config.SingularityConfiguration;
 
-public class SingularityAuthModule implements Module {
-  private final SingularityConfiguration configuration;
+public class SingularityAuthModule extends DropwizardAwareModule<SingularityConfiguration> {
 
-  public SingularityAuthModule(SingularityConfiguration configuration) {
-    this.configuration = configuration;
-  }
+  public SingularityAuthModule() {}
 
   @Override
   public void configure(Binder binder) {
-    binder.bind(SingularityAuthenticator.class).to(configuration.getAuthConfiguration().getAuthenticator().getAuthenticatorClass());
-    binder.bind(SingularityAuthDatastore.class).to(configuration.getAuthConfiguration().getDatastore().getAuthDatastoreClass());
-    binder.bind(new TypeLiteral<Optional<SingularityUser>>() {}).toProvider(SingularityAuthenticator.class);
+    Multibinder<SingularityAuthenticator> multibinder = Multibinder.newSetBinder(binder, SingularityAuthenticator.class);
+    for (SingularityAuthenticatorClass clazz : getConfiguration().getAuthConfiguration().getAuthenticators()) {
+      multibinder.addBinding().to(clazz.getAuthenticatorClass());
+    }
+
+    binder.bind(SingularityAuthFeature.class);
+    binder.bind(SingularityMultiMethodAuthenticator.class);
+    binder.bind(SingularityAuthDatastore.class).to(getConfiguration().getAuthConfiguration().getDatastore().getAuthDatastoreClass());
     binder.bind(SingularityAuthorizationHelper.class).in(Scopes.SINGLETON);
+
+    getEnvironment().jersey().register(SingularityAuthFeature.class);
```

Comment:
Do you need this? I think that binding it should be enough.

Reply:
I can try it. I was under the impression i needed both

Reply:
you are correct, not needed. Now removed